### PR TITLE
Added pre upgrade test in end2end

### PR DESCRIPTION
**Pull request by evalsocket**
- Added pre upgrade test input for flyteadmin, It's optional for other component 

Fix: https://github.com/flyteorg/flyte/issues/2077

### CODE REVIEW

File: .github/workflows/end2end.yml [hunk 1]
```diff
@@ -52,8 +49,17 @@ jobs:
           project: flytesnacks
           version: "latest"
           domain: development
-      - name: Run tests
-        id: run-tests
+      - name: Pre Upgrade Tests
+        env:
+          PRIORITIES: "${{ inputs.priorities }}"
+        run: |
+          ./boilerplate/flyte/end2end/end2end.sh
```
Comment (pmahindrakar-oss) [inline .github/workflows/end2end.yml:56]: Can you point me where this file is in github repo.

**Comment by pmahindrakar-oss** [comment]
Also can you help in mapping these requirements with the steps mentioned in the ticket and which portions cover these briefly. Want to make sure we cover all or if some are missing then we can add new test for those.

-We deploy flyte using current version n
-Register and create executions on this version n
- upgrade flyte to n+1
- Flyte deployment should be healthy
- Do basic Get /UPDATE/DELETE operations of older flyte entities
- Create new executions and perform the same GET/UPDATE/DELETE operations and additional CREATE too

File: .github/workflows/end2end.yml (hunk 1 above)
Comment (evalsocket) [reply .github/workflows/end2end.yml:56]: https://github.com/flyteorg/flyteadmin/blob/master/boilerplate/flyte/end2end/end2end.sh

**Comment by evalsocket** [comment]
@pmahindrakar-oss Please check the workflow logs https://github.com/flyteorg/flyteadmin/runs/5252483827?check_suite_focus=true

**Comment by evalsocket** [comment]
@pmahindrakar-oss That's is not available, Currently functional test it self fetch the execution while they are running, But did not fetch specific execution after upgrade, These test we can add later in functional test script.

**Comment by evalsocket** [comment]
cc: @eapolinario

**Comment by pmahindrakar-oss** [comment]
Ok.  @evalsocket do you mind creating a separate issue and link with parent. Can be a simple one where we call listExecutions and then iterate over each execution to fetch individual details in the post upgrade test. This can be named as CheckBackwardCompatTest and we can add more scenarios in this one for other entities.

**Review by pmahindrakar-oss — APPROVED** [review]

**Review by pmahindrakar-oss — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]

**Pull request merged**